proto/examples/sasltest/main.rs

```diff
@@ -0,0 +1,167 @@
+use core::panic;
+use futures_util::future::err;
+use sspi::builders::EmptyInitializeSecurityContext;
+use sspi::AuthIdentity;
+use sspi::ClientRequestFlags;
+use sspi::CredentialUse;
+use sspi::DataRepresentation;
+use sspi::Ntlm;
+use sspi::SecurityBuffer;
+use sspi::SecurityBufferType;
+use sspi::SecurityStatus;
+use sspi::Sspi;
+use sspi::SspiImpl;
+use sspi::Username;
+use std::env;
+use std::net::SocketAddr;
+use std::str::FromStr;
+use tokio::net::TcpStream;
+use tokio_util::codec::Framed;
+use tracing::Level;
+
+use futures_util::sink::SinkExt;
+use futures_util::stream::StreamExt;
+
+use ldap3_proto::proto::*;
+use ldap3_proto::LdapCodec;
+
+#[tokio::main]
+async fn main() -> Result<(), ()> {
+    let subs = tracing_subscriber::FmtSubscriber::builder()
+        .with_max_level(Level::DEBUG)
+        .finish();
+    tracing::subscriber::set_global_default(subs).expect("setting default subscriber failed");

```

Comment:
Nitpick: if using `anyhow`, it’s a bit more clean to use `.context()` instead of `.expect()` in this context, but not a big deal for an example

Reply:
removed anyhow